projects/laji/src/app/shared/api/WarehouseApi.ts

```diff
@@ -195,6 +196,13 @@ export class WarehouseApi {
     return this.warehouseQueryGet(target, query, aggregateBy, orderBy, pageSize, page, geoJSON, onlyCount);
   }
 
+  // query to unit/aggregate endpoint with front-end cache
+  @cacheReturnObservable(120000)
+  public warehouseQueryAggregateGetCached(query: WarehouseQueryInterface = {}, aggregateBy?: Array<string>, orderBy?: Array<string>, pageSize?: number, page?: number, geoJSON?: boolean, onlyCount?: boolean): Observable<PagedResult<any>|any> {
```

Comment:
I think you should cache `getGatheringCounts$` directly instead. Different uses of `warehouseQueryAggregateGet` may require different cache times

Reply:
Reason why I failed at first was the all taxaParam which caused same query to be called with different params -> no cache hit. Also taxon and year could be either string of undefined, so I tried to address this issue in the latest commit.